include/object-with-language-interface.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * @package Polylang
+ */
+
+/**
+ * Interface to use for object types that support at least one language.
+ *
+ * @since 3.3
+ */
+interface PLL_Object_With_Language_Interface {
+
+	/**
+	 * Returns the language taxonomy name.
+	 *
+	 * @since 3.3
+	 *
+	 * @return string
+	 *
+	 * @phpstan-return non-empty-string
+	 */
+	public function get_tax_language();
+
+	/**
+	 * Adds hooks.
+	 *
+	 * @since 3.3
+	 *
+	 * @return void
+	 */
+	public function init();
+
+	/**
+	 * Stores the object's language in the database.
+	 *
+	 * @since 3.3
+	 *
+	 * @param int                     $id   Object ID.
+	 * @param int|string|PLL_Language $lang Language (term_id, slug, or object).
+	 * @return void
```

Comment:
Maybe we could return a boolean to validate language assignation?

Reply:
Done in [6d0a7b4](https://github.com/polylang/polylang/pull/1149/commits/6d0a7b4c742ce9d952a103ea07c747c31984a413).